common-utils/src/main/java/eu/arrowhead/common/service/validation/PageValidator.java

```diff
@@ -30,8 +32,17 @@ public void validatePageParameter(final PageDTO page, final List<String> availab
 		logger.debug("validatePageParameter started...");
 
 		if (page != null) {
-			if (page.page() == null && page.size() != null) {
-				throw new InvalidParameterException("If size parameter is defined then page parameter cannot be undefined", origin);
+
+			if (page.size() != null) {
+
+				if (page.page() == null) {
+					throw new InvalidParameterException("If size parameter is defined then page parameter cannot be undefined", origin);
+				}
+
+				if (page.size() > maxPageSize) {
+					throw new InvalidParameterException("The page size cannot be larger than " + maxPageSize, origin);
+				}
+
```

Comment:
This is good. But you have to modify PageService as well. Whenever we use Integer.MAX_SIZE in the code we should use this maxPageSize instead.